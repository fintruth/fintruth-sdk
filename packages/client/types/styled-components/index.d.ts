import 'styled-components'

declare module 'styled-components' {
  export interface Body {
    color: string
    fontFamily: string
    fontWeight: number
    lineHeight: number
  }

  export interface Code {
    backgroundColor: string
    color: string
    fontFamily: string
    fontSize: string
    fontWeight: number
    padding: string
  }

  export interface DefaultTheme extends Variables {
    body: Body
    code: Code
    hr: Hr
    html: Html
    pre: Pre
    strong: Strong
  }

  export interface Hr {
    backgroundColor: string
    height: string
    margin: string
  }

  export interface Html {
    backgroundColor: string
    fontSize: string
    overflowX: string
    overflowY: string
    textRendering: string
  }

  export interface Pre {
    backgroundColor: string
    color: string
  }

  export interface Strong {
    color: string
    fontWeight: number
  }

  export interface Variables {
    backgroundColor: string
    black: string
    blackBis: string
    blackTer: string
    blue: string
    blueContrast: string
    borderColor: string
    borderHoverColor: string
    cyan: string
    cyanContrast: string
    danger: string
    dangerContrast: string
    darkContrast: string
    fontFamilyCode: string
    fontFamilyMonospace: string
    fontFamilyPrimary: string
    fontFamilySansSerif: string
    fontFamilySecondary: string
    gap: number
    gray: string
    grayDark: string
    grayDarker: string
    grayLight: string
    grayLighter: string
    green: string
    greenContrast: string
    info: string
    infoContrast: string
    lightContrast: string
    linkActiveBorderColor: string
    linkActiveColor: string
    linkColor: string
    linkColorContrast: string
    linkFocusBorderColor: string
    linkFocusColor: string
    linkHoverBorderColor: string
    linkHoverColor: string
    linkVisitedColor: string
    orange: string
    orangeContrast: string
    primary: string
    primaryContrast: string
    purple: string
    purpleContrast: string
    red: string
    redContrast: string
    success: string
    successContrast: string
    textColor: string
    textColorContrast: string
    textLightColor: string
    textRendering: string
    textSelectionColor: string
    textStrongColor: string
    turquoise: string
    turquoiseContrast: string
    viewportExtraLarge: number
    viewportLarge: number
    viewportMedium: number
    viewportSmall: number
    warning: string
    warningContrast: string
    white: string
    whiteBis: string
    whiteTer: string
    yellow: string
    yellowContrast: string
  }
}
