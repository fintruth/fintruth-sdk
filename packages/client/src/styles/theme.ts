import { getLuminance, hsl, hsla, rem } from 'polished'
import { Color, ColorContrast, DefaultTheme, Variant } from 'styled-components'

const contrastColor = (color: string) =>
  getLuminance(color) > 0.55 ? hsla(0, 0, 0, 0.7) : hsl(0, 0, 1)

const grayDarker = hsl(0, 0, 0.21)
const grayDark = hsl(0, 0, 0.29)
const gray = hsl(0, 0, 0.48)
const grayLight = hsl(0, 0, 0.71)
const grayLighter = hsl(0, 0, 0.86)
const grayLightest = hsl(0, 0, 0.93)

const whiteTer = hsl(0, 0, 0.96)

const light = whiteTer
const dark = grayDarker

const orange = hsl(14, 1, 0.53)
const yellow = hsl(48, 1, 0.67)
const green = hsl(141, 0.71, 0.48)
const turquoise = hsl(171, 1, 0.41)
const cyan = hsl(204, 0.86, 0.53)
const blue = hsl(217, 0.71, 0.53)
const purple = hsl(271, 1, 0.71)
const red = hsl(348, 1, 0.61)

const blueContrast = contrastColor(blue)

const primary = turquoise
const info = cyan
const success = green
const warning = yellow
const danger = red

const backgroundColor = whiteTer

const textColor = grayDark

const fontFamilySansSerif =
  'BlinkMacSystemFont, -apple-system, "Segoe UI", "Roboto", "Oxygen", "Ubuntu", "Cantarell", "Fira Sans", "Droid Sans", "Helvetica Neue", "Helvetica", "Arial", sans-serif'
const fontFamilyMonospace = 'monospace'

const gap = 32

const theme: DefaultTheme = {
  black: hsl(0, 0, 0.04),
  blackBis: hsl(0, 0, 0.07),
  blackTer: hsl(0, 0, 0.14),

  grayDarker,
  grayDark,
  gray,
  grayLight,
  grayLighter,
  grayLightest,

  whiteTer,
  whiteBis: hsl(0, 0, 0.98),
  white: hsl(0, 0, 1),

  light,
  dark,

  lightContrast: dark,
  darkContrast: light,

  orange,
  yellow,
  green,
  turquoise,
  cyan,
  blue,
  purple,
  red,

  orangeContrast: contrastColor(orange),
  yellowContrast: contrastColor(yellow),
  greenContrast: contrastColor(green),
  turquoiseContrast: contrastColor(turquoise),
  cyanContrast: contrastColor(cyan),
  blueContrast,
  purpleContrast: contrastColor(purple),
  redContrast: contrastColor(red),

  primary,
  info,
  success,
  warning,
  danger,

  primaryContrast: contrastColor(primary),
  infoContrast: contrastColor(info),
  successContrast: contrastColor(success),
  warningContrast: contrastColor(warning),
  dangerContrast: contrastColor(danger),

  backgroundColor,

  borderColor: grayLighter,
  borderHoverColor: grayLight,
  borderLightColor: grayLightest,
  borderLightHoverColor: grayLight,

  textColor,
  textColorContrast: contrastColor(textColor),
  textLightColor: gray,
  textStrongColor: grayDarker,
  textSelectionBackgroundColor: hsl(213, 0.92, 0.85),

  codeColor: red,
  codeBackgroundColor: backgroundColor,

  preColor: textColor,
  preBackgroundColor: backgroundColor,

  linkColor: blue,
  linkColorContrast: blueContrast,
  linkVisitedColor: purple,

  linkHoverColor: grayDarker,
  linkHoverBorderColor: grayLight,

  linkFocusColor: grayDarker,
  linkFocusBorderColor: blue,

  linkActiveColor: grayDarker,
  linkActiveBorderColor: grayDark,

  fontFamilySansSerif,
  fontFamilyMonospace,
  textRendering: 'optimizeLegibility',

  fontFamilyPrimary: fontFamilySansSerif,
  fontFamilySecondary: fontFamilySansSerif,
  fontFamilyCode: fontFamilyMonospace,

  blockSpacing: rem(24),

  gap,
  small: 769,
  medium: 960 + 2 * gap,
  large: 1152 + 2 * gap,
  extraLarge: 1344 + 2 * gap,

  timingFunction: 'ease-out',
  borderRadiusSmall: '2px',
  borderRadius: '4px',
  borderRadiusLarge: '6px',
  borderRadiusRounded: '999999px',
  duration: '86ms',
}

export const variantColors: Record<Variant, Color> = {
  danger: 'danger',
  dark: 'dark',
  info: 'info',
  light: 'light',
  primary: 'primary',
  success: 'success',
  warning: 'warning',
}

export const variantColorContrasts: Record<Variant, ColorContrast> = {
  danger: 'dangerContrast',
  dark: 'darkContrast',
  info: 'infoContrast',
  light: 'lightContrast',
  primary: 'primaryContrast',
  success: 'successContrast',
  warning: 'warningContrast',
}

export default theme
