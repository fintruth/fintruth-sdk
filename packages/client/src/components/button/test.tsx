import React from 'react'
import { render } from 'react-testing-library'
import { ThemeProvider } from 'styled-components'

import theme from 'styles/theme'
import Button from '.'

test('should render correctly', () => {
  const { container } = render(
    <ThemeProvider theme={theme}>
      <Button>child</Button>
    </ThemeProvider>
  )

  expect(container.firstChild).toMatchSnapshot()
})

test('should render correctly when inverted', () => {
  const { container } = render(
    <ThemeProvider theme={theme}>
      <Button isInverted>child</Button>
    </ThemeProvider>
  )

  expect(container.firstChild).toMatchSnapshot()
})

test('should render correctly when outlined', () => {
  const { container } = render(
    <ThemeProvider theme={theme}>
      <Button isOutlined>child</Button>
    </ThemeProvider>
  )

  expect(container.firstChild).toMatchSnapshot()
})

test('should render correctly when inverted and outlined', () => {
  const { container } = render(
    <ThemeProvider theme={theme}>
      <Button isInverted isOutlined>
        child
      </Button>
    </ThemeProvider>
  )

  expect(container.firstChild).toMatchSnapshot()
})

test('should render the primary variant correctly', () => {
  const { container } = render(
    <ThemeProvider theme={theme}>
      <Button variant="primary">child</Button>
    </ThemeProvider>
  )

  expect(container.firstChild).toMatchSnapshot()
})

test('should render the primary variant correctly when inverted', () => {
  const { container } = render(
    <ThemeProvider theme={theme}>
      <Button isInverted variant="primary">
        child
      </Button>
    </ThemeProvider>
  )

  expect(container.firstChild).toMatchSnapshot()
})

test('should render the primary variant correctly when outlined', () => {
  const { container } = render(
    <ThemeProvider theme={theme}>
      <Button isOutlined variant="primary">
        child
      </Button>
    </ThemeProvider>
  )

  expect(container.firstChild).toMatchSnapshot()
})

test('should render the primary variant correctly when inverted and outlined', () => {
  const { container } = render(
    <ThemeProvider theme={theme}>
      <Button isInverted isOutlined variant="primary">
        child
      </Button>
    </ThemeProvider>
  )

  expect(container.firstChild).toMatchSnapshot()
})

test('should render the danger variant correctly', () => {
  const { container } = render(
    <ThemeProvider theme={theme}>
      <Button variant="danger">child</Button>
    </ThemeProvider>
  )

  expect(container.firstChild).toMatchSnapshot()
})

test('should render the danger variant correctly when inverted', () => {
  const { container } = render(
    <ThemeProvider theme={theme}>
      <Button isInverted variant="danger">
        child
      </Button>
    </ThemeProvider>
  )

  expect(container.firstChild).toMatchSnapshot()
})

test('should render the danger variant correctly when outlined', () => {
  const { container } = render(
    <ThemeProvider theme={theme}>
      <Button isOutlined variant="danger">
        child
      </Button>
    </ThemeProvider>
  )

  expect(container.firstChild).toMatchSnapshot()
})

test('should render the danger variant correctly when inverted and outlined', () => {
  const { container } = render(
    <ThemeProvider theme={theme}>
      <Button isInverted isOutlined variant="danger">
        child
      </Button>
    </ThemeProvider>
  )

  expect(container.firstChild).toMatchSnapshot()
})
