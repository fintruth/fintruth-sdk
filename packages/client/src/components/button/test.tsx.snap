// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  border: 1px solid transparent;
  border-radius: 4px;
  box-shadow: none;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-size: 1rem;
  height: 2.25em;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  line-height: 1.5;
  padding: calc(0.375em - 1px) calc(0.625em - 1px);
  position: relative;
  vertical-align: top;
  -webkit-touch-callout: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  border-width: 1px;
  cursor: pointer;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: calc(0.375em - 1px) 0.75em;
  text-align: center;
  white-space: nowrap;
  background-color: #fff;
  border-color: #dbdbdb;
  color: #363636;
}

.c0:focus,
.c0:active {
  outline: none;
}

.c0[disabled],
fieldset[disabled] .c0 {
  cursor: not-allowed;
}

.c0:hover {
  border-color: #b5b5b5;
  color: #363636;
}

.c0:focus {
  border-color: #3273dc;
  color: #363636;
}

.c0:active {
  border-color: #4a4a4a;
  color: #363636;
}

.c0[disabled],
fieldset[disabled] .c0 {
  background-color: #fff;
  border-color: #dbdbdb;
  color: #363636;
  opacity: 0.5;
}

.c0:focus:not(:active):enabled {
  box-shadow: 0 0 0 0.125em rgba(50,115,220,0.25);
}

.c0 strong {
  color: inherit;
}

<button
  class="c0"
>
  child
</button>
`;

exports[`should render correctly when inverted 1`] = `
.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  border: 1px solid transparent;
  border-radius: 4px;
  box-shadow: none;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-size: 1rem;
  height: 2.25em;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  line-height: 1.5;
  padding: calc(0.375em - 1px) calc(0.625em - 1px);
  position: relative;
  vertical-align: top;
  -webkit-touch-callout: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  border-width: 1px;
  cursor: pointer;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: calc(0.375em - 1px) 0.75em;
  text-align: center;
  white-space: nowrap;
  background-color: #fff;
  border-color: #dbdbdb;
  color: #363636;
}

.c0:focus,
.c0:active {
  outline: none;
}

.c0[disabled],
fieldset[disabled] .c0 {
  cursor: not-allowed;
}

.c0:hover {
  border-color: #b5b5b5;
  color: #363636;
}

.c0:focus {
  border-color: #3273dc;
  color: #363636;
}

.c0:active {
  border-color: #4a4a4a;
  color: #363636;
}

.c0[disabled],
fieldset[disabled] .c0 {
  background-color: #fff;
  border-color: #dbdbdb;
  color: #363636;
  opacity: 0.5;
}

.c0:focus:not(:active):enabled {
  box-shadow: 0 0 0 0.125em rgba(50,115,220,0.25);
}

.c0 strong {
  color: inherit;
}

<button
  class="c0"
>
  child
</button>
`;

exports[`should render correctly when inverted and outlined 1`] = `
.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  border: 1px solid transparent;
  border-radius: 4px;
  box-shadow: none;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-size: 1rem;
  height: 2.25em;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  line-height: 1.5;
  padding: calc(0.375em - 1px) calc(0.625em - 1px);
  position: relative;
  vertical-align: top;
  -webkit-touch-callout: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  border-width: 1px;
  cursor: pointer;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: calc(0.375em - 1px) 0.75em;
  text-align: center;
  white-space: nowrap;
  background-color: #fff;
  border-color: #dbdbdb;
  color: #363636;
}

.c0:focus,
.c0:active {
  outline: none;
}

.c0[disabled],
fieldset[disabled] .c0 {
  cursor: not-allowed;
}

.c0:hover {
  border-color: #b5b5b5;
  color: #363636;
}

.c0:focus {
  border-color: #3273dc;
  color: #363636;
}

.c0:active {
  border-color: #4a4a4a;
  color: #363636;
}

.c0[disabled],
fieldset[disabled] .c0 {
  background-color: #fff;
  border-color: #dbdbdb;
  color: #363636;
  opacity: 0.5;
}

.c0:focus:not(:active):enabled {
  box-shadow: 0 0 0 0.125em rgba(50,115,220,0.25);
}

.c0 strong {
  color: inherit;
}

<button
  class="c0"
>
  child
</button>
`;

exports[`should render correctly when outlined 1`] = `
.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  border: 1px solid transparent;
  border-radius: 4px;
  box-shadow: none;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-size: 1rem;
  height: 2.25em;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  line-height: 1.5;
  padding: calc(0.375em - 1px) calc(0.625em - 1px);
  position: relative;
  vertical-align: top;
  -webkit-touch-callout: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  border-width: 1px;
  cursor: pointer;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: calc(0.375em - 1px) 0.75em;
  text-align: center;
  white-space: nowrap;
  background-color: #fff;
  border-color: #dbdbdb;
  color: #363636;
}

.c0:focus,
.c0:active {
  outline: none;
}

.c0[disabled],
fieldset[disabled] .c0 {
  cursor: not-allowed;
}

.c0:hover {
  border-color: #b5b5b5;
  color: #363636;
}

.c0:focus {
  border-color: #3273dc;
  color: #363636;
}

.c0:active {
  border-color: #4a4a4a;
  color: #363636;
}

.c0[disabled],
fieldset[disabled] .c0 {
  background-color: #fff;
  border-color: #dbdbdb;
  color: #363636;
  opacity: 0.5;
}

.c0:focus:not(:active):enabled {
  box-shadow: 0 0 0 0.125em rgba(50,115,220,0.25);
}

.c0 strong {
  color: inherit;
}

<button
  class="c0"
>
  child
</button>
`;

exports[`should render the danger variant correctly 1`] = `
.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  border: 1px solid transparent;
  border-radius: 4px;
  box-shadow: none;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-size: 1rem;
  height: 2.25em;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  line-height: 1.5;
  padding: calc(0.375em - 1px) calc(0.625em - 1px);
  position: relative;
  vertical-align: top;
  -webkit-touch-callout: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  border-width: 1px;
  cursor: pointer;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: calc(0.375em - 1px) 0.75em;
  text-align: center;
  white-space: nowrap;
  background-color: #ff3860;
  border-color: transparent;
  color: #fff;
}

.c0:focus,
.c0:active {
  outline: none;
}

.c0[disabled],
fieldset[disabled] .c0 {
  cursor: not-allowed;
}

.c0:hover {
  background-color: #ff2b56;
}

.c0:active {
  background-color: #ff1f4c;
}

.c0[disabled],
fieldset[disabled] .c0 {
  background-color: #ff3860;
  opacity: 0.5;
}

.c0:focus:not(:active):enabled {
  box-shadow: 0 0 0 0.125em rgba(255,56,96,0.25);
}

.c0 strong {
  color: inherit;
}

<button
  class="c0"
>
  child
</button>
`;

exports[`should render the danger variant correctly when inverted 1`] = `
.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  border: 1px solid transparent;
  border-radius: 4px;
  box-shadow: none;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-size: 1rem;
  height: 2.25em;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  line-height: 1.5;
  padding: calc(0.375em - 1px) calc(0.625em - 1px);
  position: relative;
  vertical-align: top;
  -webkit-touch-callout: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  border-width: 1px;
  cursor: pointer;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: calc(0.375em - 1px) 0.75em;
  text-align: center;
  white-space: nowrap;
  background-color: #fff;
  border-color: transparent;
  color: #ff3860;
}

.c0:focus,
.c0:active {
  outline: none;
}

.c0[disabled],
fieldset[disabled] .c0 {
  cursor: not-allowed;
}

.c0:hover {
  background-color: #f9f9f9;
}

.c0:active {
  background-color: #f2f2f2;
}

.c0[disabled],
fieldset[disabled] .c0 {
  background-color: #fff;
  opacity: 0.5;
}

.c0:focus:not(:active):enabled {
  box-shadow: 0 0 0 0.125em rgba(255,56,96,0.25);
}

.c0 strong {
  color: inherit;
}

<button
  class="c0"
>
  child
</button>
`;

exports[`should render the danger variant correctly when inverted and outlined 1`] = `
.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  border: 1px solid transparent;
  border-radius: 4px;
  box-shadow: none;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-size: 1rem;
  height: 2.25em;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  line-height: 1.5;
  padding: calc(0.375em - 1px) calc(0.625em - 1px);
  position: relative;
  vertical-align: top;
  -webkit-touch-callout: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  border-width: 1px;
  cursor: pointer;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: calc(0.375em - 1px) 0.75em;
  text-align: center;
  white-space: nowrap;
  background-color: transparent;
  border-color: #fff;
  color: #fff;
}

.c0:focus,
.c0:active {
  outline: none;
}

.c0[disabled],
fieldset[disabled] .c0 {
  cursor: not-allowed;
}

.c0:hover {
  background-color: #fff;
  color: #ff3860;
}

.c0:active {
  background-color: #f9f9f9;
  border-color: transparent;
  color: #ff3860;
}

.c0[disabled],
fieldset[disabled] .c0 {
  background-color: transparent;
  border-color: #fff;
  color: #fff;
  opacity: 0.5;
}

.c0:focus:not(:active):enabled {
  box-shadow: 0 0 0 0.125em rgba(255,255,255,0.25);
}

.c0 strong {
  color: inherit;
}

<button
  class="c0"
>
  child
</button>
`;

exports[`should render the danger variant correctly when outlined 1`] = `
.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  border: 1px solid transparent;
  border-radius: 4px;
  box-shadow: none;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-size: 1rem;
  height: 2.25em;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  line-height: 1.5;
  padding: calc(0.375em - 1px) calc(0.625em - 1px);
  position: relative;
  vertical-align: top;
  -webkit-touch-callout: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  border-width: 1px;
  cursor: pointer;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: calc(0.375em - 1px) 0.75em;
  text-align: center;
  white-space: nowrap;
  background-color: transparent;
  border-color: #ff3860;
  color: #ff3860;
}

.c0:focus,
.c0:active {
  outline: none;
}

.c0[disabled],
fieldset[disabled] .c0 {
  cursor: not-allowed;
}

.c0:hover {
  background-color: #ff3860;
  color: #fff;
}

.c0:active {
  background-color: #ff2b56;
  border-color: transparent;
  color: #fff;
}

.c0[disabled],
fieldset[disabled] .c0 {
  background-color: transparent;
  border-color: #ff3860;
  color: #ff3860;
  opacity: 0.5;
}

.c0:focus:not(:active):enabled {
  box-shadow: 0 0 0 0.125em rgba(255,56,96,0.25);
}

.c0 strong {
  color: inherit;
}

<button
  class="c0"
>
  child
</button>
`;

exports[`should render the primary variant correctly 1`] = `
.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  border: 1px solid transparent;
  border-radius: 4px;
  box-shadow: none;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-size: 1rem;
  height: 2.25em;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  line-height: 1.5;
  padding: calc(0.375em - 1px) calc(0.625em - 1px);
  position: relative;
  vertical-align: top;
  -webkit-touch-callout: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  border-width: 1px;
  cursor: pointer;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: calc(0.375em - 1px) 0.75em;
  text-align: center;
  white-space: nowrap;
  background-color: #00d1b2;
  border-color: transparent;
  color: #fff;
}

.c0:focus,
.c0:active {
  outline: none;
}

.c0[disabled],
fieldset[disabled] .c0 {
  cursor: not-allowed;
}

.c0:hover {
  background-color: #00c4a7;
}

.c0:active {
  background-color: #00b89c;
}

.c0[disabled],
fieldset[disabled] .c0 {
  background-color: #00d1b2;
  opacity: 0.5;
}

.c0:focus:not(:active):enabled {
  box-shadow: 0 0 0 0.125em rgba(0,209,178,0.25);
}

.c0 strong {
  color: inherit;
}

<button
  class="c0"
>
  child
</button>
`;

exports[`should render the primary variant correctly when inverted 1`] = `
.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  border: 1px solid transparent;
  border-radius: 4px;
  box-shadow: none;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-size: 1rem;
  height: 2.25em;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  line-height: 1.5;
  padding: calc(0.375em - 1px) calc(0.625em - 1px);
  position: relative;
  vertical-align: top;
  -webkit-touch-callout: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  border-width: 1px;
  cursor: pointer;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: calc(0.375em - 1px) 0.75em;
  text-align: center;
  white-space: nowrap;
  background-color: #fff;
  border-color: transparent;
  color: #00d1b2;
}

.c0:focus,
.c0:active {
  outline: none;
}

.c0[disabled],
fieldset[disabled] .c0 {
  cursor: not-allowed;
}

.c0:hover {
  background-color: #f9f9f9;
}

.c0:active {
  background-color: #f2f2f2;
}

.c0[disabled],
fieldset[disabled] .c0 {
  background-color: #fff;
  opacity: 0.5;
}

.c0:focus:not(:active):enabled {
  box-shadow: 0 0 0 0.125em rgba(0,209,178,0.25);
}

.c0 strong {
  color: inherit;
}

<button
  class="c0"
>
  child
</button>
`;

exports[`should render the primary variant correctly when inverted and outlined 1`] = `
.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  border: 1px solid transparent;
  border-radius: 4px;
  box-shadow: none;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-size: 1rem;
  height: 2.25em;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  line-height: 1.5;
  padding: calc(0.375em - 1px) calc(0.625em - 1px);
  position: relative;
  vertical-align: top;
  -webkit-touch-callout: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  border-width: 1px;
  cursor: pointer;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: calc(0.375em - 1px) 0.75em;
  text-align: center;
  white-space: nowrap;
  background-color: transparent;
  border-color: #fff;
  color: #fff;
}

.c0:focus,
.c0:active {
  outline: none;
}

.c0[disabled],
fieldset[disabled] .c0 {
  cursor: not-allowed;
}

.c0:hover {
  background-color: #fff;
  color: #00d1b2;
}

.c0:active {
  background-color: #f9f9f9;
  border-color: transparent;
  color: #00d1b2;
}

.c0[disabled],
fieldset[disabled] .c0 {
  background-color: transparent;
  border-color: #fff;
  color: #fff;
  opacity: 0.5;
}

.c0:focus:not(:active):enabled {
  box-shadow: 0 0 0 0.125em rgba(255,255,255,0.25);
}

.c0 strong {
  color: inherit;
}

<button
  class="c0"
>
  child
</button>
`;

exports[`should render the primary variant correctly when outlined 1`] = `
.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  border: 1px solid transparent;
  border-radius: 4px;
  box-shadow: none;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-size: 1rem;
  height: 2.25em;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  line-height: 1.5;
  padding: calc(0.375em - 1px) calc(0.625em - 1px);
  position: relative;
  vertical-align: top;
  -webkit-touch-callout: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  border-width: 1px;
  cursor: pointer;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: calc(0.375em - 1px) 0.75em;
  text-align: center;
  white-space: nowrap;
  background-color: transparent;
  border-color: #00d1b2;
  color: #00d1b2;
}

.c0:focus,
.c0:active {
  outline: none;
}

.c0[disabled],
fieldset[disabled] .c0 {
  cursor: not-allowed;
}

.c0:hover {
  background-color: #00d1b2;
  color: #fff;
}

.c0:active {
  background-color: #00c4a7;
  border-color: transparent;
  color: #fff;
}

.c0[disabled],
fieldset[disabled] .c0 {
  background-color: transparent;
  border-color: #00d1b2;
  color: #00d1b2;
  opacity: 0.5;
}

.c0:focus:not(:active):enabled {
  box-shadow: 0 0 0 0.125em rgba(0,209,178,0.25);
}

.c0 strong {
  color: inherit;
}

<button
  class="c0"
>
  child
</button>
`;
